keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,1044 @@
+controller, and has some pods created.
+Then, if the label is switched to the mirroring Kueue controller (which by
+itself does not manage pods). Then, the pods are leaking and remain running.
+
+In order to avoid the risk of pods leaking between the controllers when changing
+value of the `managed-by` label, we make the label immutable (allow to be added
+on Job creation, but fail requests trying to update its value, see also
+[label mutability](#label-mutability)).
+
+However, the question remains if we can make the label mutable when the job is
+stopped, similarly, as we do with the `AllowMutableSchedulingDirectives` flag
+which guards mutability of the Job's pod template labels.
+
+It seems possible, and could be handy in [Story 1](#story-1), but it is also not
+a blocker.
+
+It would also the complicate debuggability of the feature.
+
+We decide to keep the label immutable, at least for [Beta](#beta), we will
+re-evaluate the decision for [GA](#ga).
+
+### Risks and Mitigations
+
+#### Ecosystem fragmentation due to forks
+
+The mechanism to disable the main Job controller opens the door for users to
+substitute it with a fork. This may create more fragmentation in the community
+as users may prefer to use their forked controllers rather than contribute
+upstream.
+
+First, this risk, to some extent, exists even today as admins with access to the
+control plane can disable by passing `--controllers=-job,*` in the manifest for
+`kube-controller-manager` (see more info on the `--controllers` flag
+[here](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/)).
+
+Second, we believe that users who had the need to fork the Job controller
+already introduced dedicated Job CRDs for their needs.
+
+#### Two controllers running at the same time on old version
+
+It is possible that one configures MultiKueue (or another project) with an
+older version of k8s which does not support the label yet. In that case
+two controllers might start running and compete with Job status updates at the
+same time.
+
+Note that an analogous situation may happen when the version of Kubernetes
+already supports the label, but the feature gate is disabled in `kube-controller-manager`.
+
+To mitigate this risk we warn about it in Kueue documentation, to use this label
+only against newer versions of Kubernetes.
+
+Finally, this risk will fade away with time as the new versions of
+Kubernetes support it.
+
+#### Debuggability
+
+With this mechanism new failure modes can occur. For example, a user may make
+a typo in the label value, or the cluster administrator may not install the
+custom controller (like MultiKueue) on that cluster.
+
+In such cases the user may not observe any progress by the job for a long time
+and my need to debug the Job.
+
+In order to allow for debugging of situations like this the Job controller will
+put a log line indicating the synchronization is delegated to another controller
```

Comment:
Maybe event/condition (or both) something more visible to a user would be better. Log lines are only accessible to cluster administrators, although it should still log that information.

Reply:
Indeed, the point was already raised here: https://github.com/kubernetes/enhancements/pull/4370#discussion_r1453688327. I would suggest to keep the discussion in one thread.